src/System.Management.Automation/namespaces/FileSystemContentStream.cs

```diff
@@ -843,7 +843,9 @@ private void CreateStreams(string filePath, string streamName, FileMode fileMode
             {
                 if (!String.IsNullOrEmpty(streamName))
                 {
+#if !UNIX
```

Comment:
On UNIX, _stream will be null causing a ArgumentNullException from new StreamReader at line 874.  At best, it will be misleading. Consider throwing NotSupportException on UNIX in the associated constructors